packages/build-deb-packages.sh

```diff
@@ -42,55 +42,65 @@ fi
 FULL_VERSION="${VERSION}-${DEBIAN_VERSION}"
 
 tmpdir=`mktemp -d /tmp/gdr.XXXXXX`
-if [ ! -d "$tmpdir" ]; then
+if [ ! -d "${tmpdir}" ]; then
     echo "Failed to create a temp directory!" >&2
     exit 1
 fi
 
 echo "Building gdrcopy debian packages version ${FULL_VERSION} ..."
 
-echo "Working in $tmpdir ..."
+echo "Working in ${tmpdir} ..."
 
 ex cd ${TOP_DIR_PATH}
 
-ex mkdir -p $tmpdir/gdrcopy
-ex rm -rf $tmpdir/gdrcopy/*
-ex cp -r Makefile README.md include src tests LICENSE config_arch packages/debian $tmpdir/gdrcopy/
-ex cp README.md $tmpdir/gdrcopy/debian/README.Debian
-ex cp README.md $tmpdir/gdrcopy/debian/README.source
-ex rm -f $tmpdir/gdrcopy_${VERSION}.orig.tar.gz
+ex mkdir -p ${tmpdir}/gdrcopy
+ex rm -rf ${tmpdir}/gdrcopy/*
+ex cp -r Makefile README.md include src tests LICENSE config_arch packages/debian ${tmpdir}/gdrcopy/
+ex cp README.md ${tmpdir}/gdrcopy/debian/README.Debian
+ex cp README.md ${tmpdir}/gdrcopy/debian/README.source
+ex rm -f ${tmpdir}/gdrcopy_${VERSION}.orig.tar.gz
 
-ex cd $tmpdir/gdrcopy
-ex find . -type f -exec sed -i "s/@VERSION@/${FULL_VERSION}/g" {} +
+ex cd ${tmpdir}/gdrcopy
+ex find . -type f -exec sed -i "s/@FULL_VERSION@/${FULL_VERSION}/g" {} +
+ex find . -type f -exec sed -i "s/@VERSION@/${VERSION}/g" {} +
 
-ex cd $tmpdir
+ex cd ${tmpdir}
 ex mv gdrcopy gdrcopy-${VERSION}
 ex tar czvf gdrcopy_${VERSION}.orig.tar.gz gdrcopy-${VERSION}
 
-ex cd $tmpdir/gdrcopy-${VERSION}
+ex cd ${tmpdir}/gdrcopy-${VERSION}
 ex debuild --set-envvar=CUDA=${CUDA} --set-envvar=PKG_CONFIG_PATH=${PKG_CONFIG_PATH} -us -uc
 
 echo
 echo "Building dkms module ..."
-ex cd $tmpdir/gdrcopy-${VERSION}/src/gdrdrv
+ex cd ${tmpdir}/gdrcopy-${VERSION}/src/gdrdrv
 ex make clean
 
-ex mkdir -p $tmpdir/gdrdrv-dkms-${VERSION}/
-ex cp -r $tmpdir/gdrcopy-${VERSION}/src/gdrdrv $tmpdir/gdrdrv-dkms-${VERSION}/gdrdrv-${VERSION}
-ex cp ${SCRIPT_DIR_PATH}/dkms.conf $tmpdir/gdrdrv-dkms-${VERSION}/gdrdrv-${VERSION}/
-ex cd $tmpdir/gdrdrv-dkms-${VERSION}/
+dkmsdir="${tmpdir}/gdrdrv-dkms-${VERSION}"
+ex mkdir -p ${dkmsdir}
+ex cp -r ${tmpdir}/gdrcopy-${VERSION}/src/gdrdrv ${dkmsdir}/gdrdrv-${VERSION}
+ex cp ${SCRIPT_DIR_PATH}/dkms.conf ${dkmsdir}/gdrdrv-${VERSION}/
+ex cd ${dkmsdir}
 ex cp -r ${SCRIPT_DIR_PATH}/dkms/* .
-ex find . -type f -exec sed -i "s/@VERSION@/${FULL_VERSION}/g" {} +
+ex find . -type f -exec sed -i "s/@FULL_VERSION@/${FULL_VERSION}/g" {} +
+ex find . -type f -exec sed -i "s/@VERSION@/${VERSION}/g" {} +
```

Comment:
dkmsdir is a subdir of tmpdir, are these two 'find + sed' commands superfluous ?

Reply:
No, they are not. These two commands are important in handling deb_version while keep the gdrdrv folder without deb_version.

Note that FULL_VERSION and VERSION refer to different strings.

Reply:
understood.
I referred to lines 64 and 65. But I see that you copy stuff into  ${dkmsdir}/gdrdrv-${VERSION} at line 81, which is later, so you need these extra sed.